src/aiida_icon/calculations.py

```diff
@@ -170,6 +171,14 @@ def prepare_for_submission(self, folder: folders.Folder) -> datastructures.CalcI
                     "run_icon.sh",
                 )
             )
+        if "run_env" in self.inputs:
+            calcinfo.local_copy_list.append(
+                (
+                    self.inputs.run_env.uuid,
+                    self.inputs.run_env.filename,
+                    "run_env.sh",
+                )
+            )
```

Comment:
add sourcing this file to calcinfo prepend text

Reply:
Actually, after looking at the `aiida_core` source code, it seems it might be possible. Will try.

Reply:
Indeed `preprend_text` works differently, so I implemented it right here.